src/services/edusharing/logic/connector.js

```
@@ -0,0 +1,243 @@
+		EduSharingConnector.instance = this;
+	}
+
+	static get headers() {
+		return {
+			Accept: 'application/json',
+			'Content-type': 'application/json',
+		};
+	}
+
+	static get authorization() {
+		const headers = {
+			...EduSharingConnector.headers,
+			Authorization: `Basic ${Buffer.from(`${Configuration.get('ES_USER')
+			}:${Configuration.get('ES_PASSWORD')}`).toString(
+				'base64',
+			)}`,
+		};
+
+		return headers;
+	}
+
+
+	// gets cookie (JSESSION) and attach it to header
+	getCookie() {
+		const cookieOptions = {
+			uri: `${Configuration.get('ES_DOMAIN')}/edu-sharing/rest/authentication/v1/validateSession`,
+			method: 'GET',
+			headers: EduSharingConnector.authorization,
+			resolveWithFullResponse: true,
+			json: true,
+		};
+		return request(cookieOptions)
+			.then((result) => {
+				if (
+					result.statusCode !== 200
+					|| result.body.isValidLogin !== true
+				) {
+					throw Error('authentication error with edu sharing');
+				}
+				return result.headers['set-cookie'][0];
+			})
+			.catch((err) => {
+				// eslint-disable-next-line no-console
+				console.error('error: ', err);
+			});
+	}
+
+	// gets access_token and refresh_token
+	getAuth() {
+		const oauthoptions = {
+			method: 'POST',
+			url: `${Configuration.get('ES_DOMAIN')}/edu-sharing/oauth2/token`,
+			headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+
+			// eslint-disable-next-line max-len
+			body: `grant_type=${Configuration.get('ES_GRANT_TYPE')}&client_id=${
+				Configuration.get('ES_CLIENT_ID')
+			}&client_secret=${Configuration.get('ES_OAUTH_SECRET')}&username=${
+				Configuration.get('ES_USER')
+			}&password=${Configuration.get('ES_PASSWORD')}`,
+			timeout: REQUEST_TIMEOUT,
+		};
+		return request(oauthoptions).then((result) => {
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			// eslint-disable-next-line no-console
+			console.error('Oauth failed');
+			return null;
```

as this is a promise it should reject instead of returning null. this can be done by throwing an feathers error 500 instead of logging

---

OK. The total authentication request will fail if this function fails